### PR TITLE
Add AWSVPC

### PR DESCRIPTION
**Pull request by brb**
Please be aware that all the necessary enhancements are listed in https://github.com/weaveworks/weave/issues/2322.

### CODE REVIEW

File: ipam/ring/ring_test.go [hunk 1]
```diff
@@ -540,11 +544,11 @@ func TestClaimForPeers(t *testing.T) {
 	// Different end to usual so we get a number of addresses that a)
 	// is smaller than the max number of peers, and b) is divisible by
 	// some number of peers. This maximises coverage of edge cases.
-	end := dot10
+	end := dot8
```
Comment (rade) [inline ipam/ring/ring_test.go]: why the change?

Comment (bboreham) [inline ipam/ring/ring_test.go]: We only support rings with power of two size now.

File: prog/weaveutil/attach.go [hunk 2]
```diff
@@ -20,25 +17,17 @@ func attach(args []string) error {
...
-		args = args[1:]
-	}
-	if isAWSVPC {
-		withMulticastRoute = false
+	for i := 0; i < len(args); i++ {
+		switch args[i] {
+		case "--no-multicast-route":
+			withMulticastRoute = false
+			args = append(args[:i], args[i+1:]...)
+		case "--keep-tx-on":
```
Comment (rade) [inline prog/weaveutil/attach.go]: needs to be added to usage.

File: weave [hunk 3]
```diff
@@ -894,6 +893,8 @@ container_in_host_ns() {
 attach() {
     ATTACH_ARGS=""
     [ -n "$NO_MULTICAST_ROUTE" ] && ATTACH_ARGS="--no-multicast-route"
+    # Relying on AWSVPC being set in 'ipam_cidrs allocate', except for 'weave restart'
+    [ -n "$AWSVPC" ] && ATTACH_ARGS="--no-multicast-route --keep-tx=on"
```
Comment (brb) [inline weave]: you probably meant `--keep-tx-on`

File: weave [hunk 4]
```diff
@@ -1285,6 +1286,13 @@ ipam_cidrs() {
         allocate)
             METHOD=POST
             CHECK_ALIVE="?check-alive=true"
+            # If we get an error here, we skip applying necessary steps for AWSVPC, because
+            # "attach" should work without the weave router running.
+            [ "$(call_weave GET /ipinfo/tracker)" != "awsvpc" ] || AWSVPC=1
```
Comment (brb) [inline weave]: maybe move to a separate function, because there exist two instances of the same check and set.

File: weave (hunk 3 above)
Comment (bboreham) [inline weave]: good spot! fixed.

File: weave (hunk 4 above)
Comment (bboreham) [inline weave]: done.

File: prog/weaveutil/attach.go (hunk 2 above)
Comment (bboreham) [inline prog/weaveutil/attach.go]: `--keep-tx-on` added to usage
